awx/api/serializers.py

```diff
@@ -3338,114 +3436,86 @@ def get_survey_enabled(self, obj):
-        errors = {}
-        obj = self.context.get('obj')
-        data = self.context.get('data')
+        template = self.context.get('template')
 
-        for field in obj.resources_needed_to_start:
-            if not (attrs.get(field, False) and obj._ask_for_vars_dict().get(field, False)):
-                errors[field] = _("Job Template '%s' is missing or undefined.") % field
+        template._is_manual_launch = True  # TODO: hopefully remove this
+        accepted, rejected, errors = template._accept_or_ignore_job_kwargs(**attrs)
+        self._ignored_fields = rejected
 
-        if obj.inventory and obj.inventory.pending_deletion is True:
+        if template.inventory and template.inventory.pending_deletion is True:
             errors['inventory'] = _("The inventory associated with this Job Template is being deleted.")
+        elif 'inventory' in accepted and accepted['inventory'].pending_deletion:
+            errors['inventory'] = _("The provided inventory is being deleted.")
 
-        extra_vars = attrs.get('extra_vars', {})
-        try:
-            extra_vars = parse_yaml_or_json(extra_vars, silent_failure=False)
-        except ParseError as e:
-            # Catch known user variable formatting errors
-            errors['extra_vars'] = str(e)
-
-        if self.get_survey_enabled(obj):
-            validation_errors = obj.survey_variable_validation(extra_vars)
-            if validation_errors:
-                errors['variables_needed_to_start'] = validation_errors
-
-        # Prohibit credential assign of the same CredentialType.kind
-        # Note: when multi-vault is supported, we'll have to carve out an
-        # exception to this logic
+        # Prohibit providing multiple credentials of the same CredentialType.kind
+        # or multiples of same vault id
         distinct_cred_kinds = []
-        for cred in data.get('credentials', []):
-            cred = Credential.objects.get(id=cred)
-            if cred.credential_type.pk in distinct_cred_kinds:
+        for cred in accepted.get('credentials', []):
+            if cred.unique_hash() in distinct_cred_kinds:
                 errors['credentials'] = _('Cannot assign multiple %s credentials.' % cred.credential_type.name)
-            distinct_cred_kinds.append(cred.credential_type.pk)
+            distinct_cred_kinds.append(cred.unique_hash())
 
-        # Special prohibited cases for scan jobs
-        errors.update(obj._extra_job_type_errors(data))
+        # verify that credentials (either provided or existing) don't
+        # require launch-time passwords that have not been provided
+        if 'credentials' in accepted:
+            launch_credentials = accepted['credentials']
+        else:
+            launch_credentials = template.credentials.all()
+        passwords = attrs.get('credential_passwords', {})  # get from original attrs
+        passwords_lacking = []
+        for cred in launch_credentials:
+            if cred.passwords_needed:
```

Comment:
Can we lose this `if`?  Seems like `passwords_needed` will fall back to just being an empty list.